### PR TITLE
Papa board pic18

### PR DESCRIPTION
**Pull request by DanielShteinbok**
I made redid the power based on our requirements, using SOIC instead of QFN IC for the things

---
This change is [Reviewable](https://reviewable.io/reviews/waterloo-rocketry/payload-2020/8)

### CODE REVIEW

**Comment by DanielShteinbok** [comment]
Could somebody please review the recent changes to the schematic?

**Review by LanaTomlin — CHANGES_REQUESTED** [review]
1. we should be adding +12V back to mama connector
2. We need to put 3 battery connectors and 12v regulators back
3. footprint for new dc/dc boi isn't showing up for me, can someone else check if it works for them?
4.  Need to add fuses for our batteries
5. In terms of the TPS5633DDC, we'll save ourselves a lot of trouble if we use a fixed output converter. These guys are finicky.

**Comment by wendi-yu** [comment]
Just a quick note but don't forget to squash the commits down whenever y'all merge this haha

**Comment by zx100x100** [comment]
Nice work :fantasticstuff: a few comments:
1. I think this routing could be cleaner, could just be me but I found it really hard to follow these SPI connections. Also I don't really understand what's going on. can you add more labeling to the schematic and also let me know what happening here?

2. the diode needs to be annotated, and the footprint for the regulator isn't working right (but guessing that is an issue on my end) also there tends to be a fair number of calcs for setting up regulators like this, can you add some notes on the math and parts selection? as is It's not clear what the output voltage is (from context looks to be 5V but really not obvious)
(all of this being said I agree with Lana that a fix reg would probably be better)
image

3. the SM74611 is a hecking cool part. it is however a little pricy (5$ isn't awful but it starts to add up) it's also giant and designed for really high current and as such a bit of a pain to solder. do we really need diodes? (this is dependent on the exact payload battery pack design which I'm not really familiar with) and if so can we get something a bit smaller and cheaper?

4. pretty nit picky but these net labels being right beside the pic with no wire looks kind of messy to me, I think just pulling them out a bit would help a lot.

5. what is this altimeter project thing? 
image

**Review by zx100x100 — CHANGES_REQUESTED** [review]
whoops that was a comment not review :picard_facepalm:

**Review by LanaTomlin — CHANGES_REQUESTED** [review]
YAY for fixed 5V converter! 
Couple things that still need to happen:
I think we're using 9 lipo cells (so 3 12V connections)
And we're also gonna need to put a regulator on those because Lipos won't always give us 12V

**Comment by LanaTomlin** [comment]
SO um sorry to change the power supplies again, Artem found some that'll work for both 5V and 12V regulation
(TPSM84212 for 12V and TPSM84205 for 5V) that have got pretty heckin' minimal power draw so yay!

**Review by zx100x100 — CHANGES_REQUESTED** [review]
beholdith my review :revolution: 

1. Please delete pic_and_can_controllers.pro, it is confusing and should not exist.

2. what is D1 for? the only thing I can think of is polarity protection, but you should be using a keyed connector anyways so it isn't worth this big 5$ part that will introduce extra noise into the voltage rails since it is a "smart diode"
image
3. Is JP1 actually useful for something? seems like another part that can fail, and you can just unplug the battery to do the same thing can you not?

4. depending on the mech design, it might be nice to have a connector for each of the cells, and put them in series on the board. this way you don't need to do a harness for the batteries, and if you need to replace a cell you just unplug it instead of potentially needing to re-crimp a harness or something. 
If you don't want to do this can you explain  to me how we are connecting the batteries together on slack?

5. you are using the +3.3VA symbol for your 11-16V rail, and just not showing the name. this is an excellent way to accidently design a board which shorts the microcontroller to 14V. please use something like +BATT instead 
image

6. would also mean you wouldn't need to do things like this, could instead use the +BATT symbol wherever battery_power is used.
image
Also you don't need to do hierarchical net labels for ground or any other power symbol. All the ground symbols in the entire project are one net. this is done a few other places, please fix it everywhere.

7. unessisairly scavenger hunt here, just put the hierarchical net label where you want it
image

8. I would reduce the unnecessary losses and load on U3 by attaching the boost converter (U2) directly to +BATT. U2 is a voltage regulator, it doesn't care if the input varies over time.

9. while I'm at it, the reference for the boost converter is hidden, please unhide it, it's important, had to figure out it was U2 by hitting edit.

10. webench wants a login. means I can't check the component selection for U2

11. U3 needs PWR_EN high to turn on, but this will turn off the 5V rail when PWR_EN is high. I think you need to add an NMOS to this circuit to fix that, while you are at it remember we already have lots of NMOS and PMOSs, so pick a part already used on another board if you can.
image

12. given there already is a 12V rail, Lets complete the Payload CAN bus and attach pin 1 to 12V 
image

13. Please note the cutoff frequency of the low pas filters on the schematic

14. +V SiPM isn't attached to anything?

15. I mentioned it previously I think, but the Rocket CAN bus is mirrored from what it should be. 
image

Overall looking pretty solid though :fantasticstuff:

**Review by zx100x100 — CHANGES_REQUESTED** [review]
back at it again with the reviews :bordy_blub:

1. Please add .sch-bak to the gitignore, lots of mildly irritating backup files rn

2. Super minor nitpic, but since the connections are already well labeled it's easier to read if you can just use wires.
image

3. dig the PDF of the webench designer, good way to document. however things like Cout2 aren't really a valid reference designator. It isn't the end of the world, but I'd prefer you just put a text note describing what each part is for vs using the reference designator for that
image

4. Fix 5V rail power control circuit. See slack discussion in #payload on 2021-05-19
5. replace single battery connector with 4 in series per discussion on slack in same time and place as above.

6. super minor, but might as well make this 1 thing: 
image

7. this will always report 0 current, because that is the truth, there is no current flow across R11. 
image
also why are you running U2 at 3.3V when you have a 5V pick? this will work, but limit your scale to 3.3V, instead of the 4.096V of the PIC's fixed voltage reference (FVR). 

8. Item 12 of last review was not addressed

9. currently you are running 5V logic into the SD card, it will die, which is sad :sadparrot:
the classy way to fix this is to use a level shifter like on the live telemetry breakout board.
the redneck way to fix this is to use a couple voltage dividers and abuse the fact that 5V logic should technically still recognize 3.3V as a 1, as is done on the RLCS powerboard shield.

10. there needs to be an individual chip select line for each SPI device, this plays into a bigger problem.

11.you are pulling the PIC and MCP2515 symbols from a rescue library, which as they should be, are gitignored. please change to reference a non-backup library. 
image


Overall it looks really good, these are mostly nitpics, but you have a bit of an IO problem.
you need 1 more pin for the logger CS, and you don't have it. not to mention there are no debug LEDs which is not good. Lets discuss on slack cause this is a tricky one.

**Comment by LanaTomlin** [comment]
Small potatoes we have left to fry:
1. Ensure Sleep Control logic actually works
2. Double check current sensing arrangement
3. Make sure connectors are correctly oriented
4. Some of the components are on different grids, which makes things not line up properly, figure out why the fuck that's happening & fix it
6. add actual debug LEDs ( some different colours)
7. add current sensing outputs onto MCUS
8. Add different CS lines
9. Change U2 to run off 5V
image
10. These need to have actual reference designators (C1, C2, etc) not Cout2. 
11. Remove unnecessary files from the commit

Medium potatoes to fry:
1. Update current sensing math & resistor values
2. Make actual guestimates about fuse values (probably around 400mA, idk)
3. Level shift SD card logic

Big potatoes to fry:
1. Change mama board connectors to can connectors
2. remove all ADC and Interrupt connections

**Comment by ZTeper** [comment]
image
 Why is 5V connected to 5V through a PMOS?

**Comment by ZTeper** [comment]
image

Add part numbers for these MOSFETs

**Comment by ZTeper** [comment]
image

Choose LED resistor value based on supply votlage

**Comment by zx100x100** [comment]
> image
> 
> Choose LED resistor value based on supply votlage

As an addition, do you really need 4 LEDs that are permanently on and will draw like 40-80mA, but you can't see them because they are in the CubeSat.

**Comment by ZTeper** [comment]
image

Include polyfuses to limit current

**Comment by ZTeper** [comment]
Make sure that the 5V_SLEEP net is actually connected to the 5V regulator

image

Why does MISO have a pullup? In general, this is not necessary.

**Comment by ZTeper** [comment]
image

-Why is OSC1 disconnected on the PIC?
-Why is CAN_INT pulled up?
-PIC and MCP2515 should each have their own crystals; or they can both be driven by a single external oscillator
-Do not drive the MCP2515 clock from a digital output pin on the PIC
-Use the default pins for the SPI connections on the PIC; assign SCK to RC3, etc.
-ICSPDAT/CLK should have their own dedicated pins; do not share them with the SPI interface.

**Comment by ZTeper** [comment]
Choose a more appropriate LED resistor value; maybe 330 ohms?

**Review by dwkletke — COMMENTED** [review]
I think the Zach's covered most things. Just a few additional notes:

Is there a reason to why the rescue lib is included for symbol libraries? Either way, it isn't kept on git so the lib is missing. In general, the backup files should also be cleaned up :)
image

Might be a good idea in general to detail what some of the nets are? While the net names can provide some explanation, it's not very obvious why `5V_ALWAYS_ON` exists.

I might have missed it, but where is `5V_OUT` used?
image

**Comment by ZTeper** [comment]
image

This is a power module, not a regulator. Please replace.

**Comment by zx100x100** [comment]
> image
> 
> This is a power module, not a regulator. Please replace.

Sorry what do you want replaced? just curious

**Comment by ZTeper** [comment]
Does payload have its own CAN bus? If so, where are the termination resistors?

**Comment by ZTeper** [comment]
image

Shield should be grounded.

**Comment by LanaTomlin** [comment]
> Does payload have its own CAN bus? If so, where are the termination resistors?

We're putting them on the cables as Aaron suggested in channel a while ago. Maybe we should add a note to the schematic w/that info

**Comment by LanaTomlin** [comment]
image
i think we shoudl be using 330 ohms for the LEDs, w/ LED_SMD:LED_1206_3216Metric_Pad1.42x1.75mm_HandSolder

^that image is what's on fill sensing, and i know the LEDs we have in the bay worked on that board

**Comment by LanaTomlin** [comment]
some of the custom symbols are missing pin numbers- this is incredibly important that these match the datasheet so the footprints line up

**Comment by LanaTomlin** [comment]
The AP2114H-3.3TRG1 SOT-223 package has 4 pins (big boi on gnd) not 3, symbol needs to be updated

**Comment by LanaTomlin** [comment]
TPS61175QPWPRQ1 has the wrong pin numbering - it's got HTSSOP-14 package adn this guy has 15 pins and it's missing

**Comment by JaredWatson** [comment]
Just checking that this voltage divider is intentonal. If you want the option to select PU/PD make sure to mark one as DNP.

image

nit: please include units here and on resistors use R instead of O or Ohm (or nothing)

image


Thats all I really have for now, I'd like to give it another look after some of the other comments are addressed.

**Comment by rdakhil** [comment]
Yes, the voltage divider is intentional (5V to 3.3V), with a 10k pull-up resistor. I will note that in the schematic

**Comment by JaredWatson** [comment]
> Yes, the voltage divider is intentional (5V to 3.3V), with a 10k pull-up resistor. I will note that in the schematic.
> 
> Edit: I now realize the main issue, will try to think of a solution.

What purpose does R22 serve? I get that R3 is a PU but is R22 a PD?

**Comment by rdakhil** [comment]
> > Yes, the voltage divider is intentional (5V to 3.3V), with a 10k pull-up resistor. I will note that in the schematic.
> > Edit: I now realize the main issue, will try to think of a solution.
> 
> What purpose does R22 serve? I get that R3 is a PU but is R22 a PD?

I didn't intend for it to serve as a PD, but instead R22 along with R18 were meant to serve as a voltage divider to bring the 5V logic down to 3.3V

**Comment by LanaTomlin** [comment]
1. You can add an additional ground pour on the top plane to make gnd even stronger
2. 
image
That looks like you've got the polygon set to "thermal" instead of "solid". We want "solid"
3. 
image
the traces being bigger than the pads is giving me a pre-soldering heart attack, I'd reccoment using a polygon to make big traces on little pads, so you don't overflow. See U7 for this 
image
it would look a lot nicer if you put all of the power plane vias straight off the resistors w/a consistant trace size (I reccomend 1mm bc it envelops the via and makes it look really pretty)
4. 
image
That's a funky lookin' trace, also usually try not to route under an IC for visability reasons, you could use a via to avoid that, but that's a nitpick here. Also the traces are misligned from the PIC
5. Inconsistance trace widths: C15, C16, C12, R24
6.  It would look a lot nicer if you used a polygon here 
image

7. Daisy chaining
image

8. Use a polygon instead of the 5V always on trace.
9. Got a spare GND via, and some of these fias are REALLLY close to the pads, a bit too close for comfort
image

10. via on a pad is not...ideal 'cause you'll end up with a bunch of solder in the via. Also more traces that are bigger than the pads here
image

11. Nitpick but you could make these traces look a lot nicer if there were the same size & straight
12. also got some funky traces going on here
image

13. that via is WAyyyy too close for comfort
image
14. daisy chaining, also would reccomend making that 5.3V trace straight if you can
image

15. make that 5V trace a polygon, the MISO line is daisy chained, an incosistant width and has a funky bump, It would be a bit better if you make the trace between R12 and Q1 go out and back in so it doesn't go under the mFET for visability, also it would look a lot better if those vias were centered, a consitent distance from teh pad and all had the same trace width. (Exept for mosi that can have a smallter trace width)
16. More daisy cahining on R23, R20, R21
17. make vias consistance trace width and distance around SD card
18. also im guessing you didn't run DRC cause some of these components look a *little* close
19. please use consistance trace sizes on mama board connectors
20. again here you want "solid" polygons, not thermal, and the D2 pad loks like it's daisy cahined, if that isn't specified in datasheet please don't daisy chain
image
21. why are C24 and R15 so far from the regulator???
22. try to make the angling of vias from J11 and J3 consistant

I guess for whole board:
1. Make sure the only things you're daisy chaining are decoupling capacitors
2. turn thicc traces into  polygons for better SHAPES
3. please don't daisy chain (found soem more on MCLR)
4. Make sure the trace size isn't bigger than the pad size
5. try to make power plane vias straight and w/traces of a consistant size

This is EXTREMELY impressive for your first layout +routing, so KUDOS!!! oh no OSC2 is also daisy chained
also why is there a nets named (-) and (+) what are they??? im scared of them

**Comment by LanaTomlin** [comment]
image
there's one notch and two can connectors here, we want these connectors each on the edge of their own notch

**Comment by LanaTomlin** [comment]
ignore my last comment, kyle's gonna do a mech check, but the rocketcan connector is definetly going to need to move to the edge of the board 'cuase WiRES

**Comment by kylemtam** [comment]
Did someone call the mech check police? :siren: Ahhhhh I should've done this a lot earlier before you started routing. Good news: I only saw one potential major problem; Bad news: There is probably one major problem (will explain at the end)

First a question: What are these pointy connector thingies underneath mini-sensor on Papa board? I didn't really see anywhere on the mini-sensor CAD that they plug into
what are these long bois

Next, as I mentioned Randy, could you please move all these highlighted parts to the other side of the board? My general thinking is that you want to minimize the number of tall things underneath the CAN boards, especially live telemetry cuz the payload redesign hasn't started yet so may as well not handicap whoever does that if possible. We have a solid 6 mm of empty air between Papa and the base plate to put relatively tall components on the bottom of Papa board.
send to NZ
I suggest you also put these battery connection thingies on the other side cuz they're tall boys, but is that possible (I dunno how those connection things work).
also send to nz

Also where is that Harwin vertical connector that connects Papa to the rest of the rocket?? I am probably blind (plz tell me I'm blind, I need an excuse to encourage myself to wear glasses in front of my laptop)
har har har


The major problem (or maybe it's just a major misunderstanding): what connectors are we using to connect Papa board? Assuming these are supposed to be Harwin vertical connectors, I uh messed up on the drawing: they don't fit (even just the ones on Papa board, forget the other half that screws in and not shown in CAD)
Harwin Connectors 1
I'll bring this problem to Slack to discuss.

Let me know if you have any question on anything - also excellent job btw, this board is beautiful :)

**Comment by kylemtam** [comment]
Oh one other thing that popped up in my mind: I noticed your screw holes are currently 3.2 mm in diameter - that correlates to a tight fit M3 clearance hole. Could you change that 3.4 or 3.6 mm in diameter instead? 3.45 mm is a regular clearance hole for a #4-40 screw and 3.4 mm is a regular M3 clearance hole - generally I personally prefer looser clearance holes to make tolerancing easier

**Comment by rdakhil** [comment]
Unfortunately the buck converter definitely can't be moved to the other side. The battery connectors should be fine to be moved to the other side, and after a few google searches I think the crystals should also be okay to be moved to the other side. Also those pointy connector things are jumpers, I now realize that they should be easily accessible, but I don't see any other place to put them. I would put them on the other side of the board, but I don't think 6mm would be enough space to connect and disconnect things. I'll try to think of a solution

**Comment by JaredWatson** [comment]
image
 I would caution against moving the oscillator if possible. I know it might be a little unnecessary but #readthedatasheet.

[source](https://ww1.microchip.com/downloads/en/DeviceDoc/PIC18(L)F2526K83-Data-Sheet-DS40001943C.pdf)

**Comment by rdakhil** [comment]
Routing is finished again after working with Kyle on the mech side, so I'm ready for some more reviews!

**Review by dwkletke — COMMENTED** [review]
Overall this looks really good! Just a few comments:
- [NIT]: The refs to most of the mounting holes are offset/off board -> check CrtYd layer
image
- Ideally these Harwin connectors would be on the edge of the board for various reasons (wiring, ease of access, etc.). However, it is not the end of the world if they can't be moved.
image
- TBH I should've asked this earlier, but is there a reason the Live Telemetry connector isn't a Harwin connector?
image
- I think `CANH_PAYLOAD` and `CANL_PAYLOAD` are still kinda daisy-chained through here:
image
- Also, remember to add the board dimensions!

**Comment by LanaTomlin** [comment]
1. Please remove these guys
image
2. 
image
might be a bit clearer if you move L1 to the top of the inductor, woud probably be easier to see
3. silkscreens don't show up on Vias, please move these guys & any others 
4. 
image
5. I really like the labels on the live telemetry connector
6. id make sure those vias aren't too close to the edge of the board. Is there a reason for teh 3rd GND via? And...could you center the +5.3V via? (It just bothers me) 
image

7. If you could get the CS_! line to be straght, that would be great. 
8.  You can put the +3V3 line on the VCC layer, and then you don't have to use two sets of vias on CANHL and CANHL rocket
9. even more of a nitpick than the rest of these but these GND vias look a bit randomly placed, if you coudl makes ure they're in some sort of pattern that would be great 
image
10. Trace weirdness:
image
image 
image

11. I know i said in my previous review i said you dont want thermal polygons, I was wrong you very much do want thermal polygons. So if you could fix up these guys & any others you find around
image
12. please make the +5.3V and GND traces the same size 
image

13. Doesnt' reallllYY matter 'cause it's going to be mostly covered by the gnd plane, but more inconsistant trace sizes: 
image
14. I'm ususally all for going around things for visability, but this looks kinda wonky. Your call. Also another randomly placed GND via 
image
15. Doing some sort of teardrop around your vias is probably a very good idea. 
16. thse jumpers are REALLY close to the can connectors, that looks like it's going to be very annoying to work with. might be worth it to replace those with 0 ohm resistors. 
image

17.  maybe use a polygon for +BATT here? Lots of trace size and placement inconsistancies
image

18. might want to either use a polygon or make sure those traces arent' touching 
image

19. please remove these REF circles.  
image

otherwise, this looks really good! 
(with the exception of rocketCAN connector in middle of board but that's a whole other can of worms)

**Review by QuantumManiac — CHANGES_REQUESTED** [review]
Sorry for the delay in review! Let's see what I can find...

image
A bit of trace wonkiness here, with this bit sticking out

image
This could totally be valid, but what's the reason to these two vias for the same net being beside each other?

image
Mostly a nitpick, but would be nice if this via were centered on the trace

image
If all three of these vias are required, it would be nice to pretty them up a bit by aligning them in a grid or row.

image
Same with these ground vias. Could be tweaked to align into a grid better

image
These vias also seem a tad misaligned

image
The left edge of this polygon could be straightened up if desired

image
I'm not sure if there's any merit other than _aesthetics_ for doing this, but it would be nice to have all the pour polygons have 90-degree edges i.e. square them up.

I think someone might have mentioned this before, but the placement of ground vias seems somewhat random and/or disorganized at times, especially when you've got some right beside each other that are misaligned. Take a sweep through the board and see if you can fix whatever you see.

image
Random unnecessary trace here, although I don't think it does any harm.

image
Same here 

image
Some odd traces over here

image
A teeny gap here. Might be nice to fill or get rid of 

image
Could straighten up the end of this trace a bit

Lastly, there may be some concerns regarding the clearances between components at places, particularly in the "Logger current sensing" area. Get them too close and they'll be a pain to solder when assembling so keep that in mind. A fun way to get a better idea of what you would be dealing with is by printing the PCB at a 1:1 scale on paper.

**Comment by rdakhil** [comment]
Here are the notes from Lance's review on call: https://docs.google.com/document/d/12llq2nqhfBjH9LKJQ7xSqCBEQJljRVznAs2_9iDezq4/edit?usp=sharing

**Review by zx100x100 — CHANGES_REQUESTED** [review]
overall looks quite good but a few changes

1. Having current sensing on the 3.3V rail but not the others seems kind of strange. I'm guessing this is a carry over from logger board. but especially since you could DNP most of it and use a zero ohm, it doesn't bother me too much. that being said, is there a reason for it in particular that I'm missing?
2. given this board is pretty much all digital it doesn't really need to be fixed, but I'm not happy to see your broken up power layer. the idea is usually that the power layer is held very steady by a bunch of bypass caps, and as such acts kind of like an honorary ground plane. breaking it up like this will increase your RF emissions / sensitivity, although probably not enough to matter here.
3. Actually change that definitely should happen: it looks like you don't have voltage sensing right now, that will make it very hard to know how much longer the payload can run for, and I'm guessing you will definitely want to be able to tell.
4. It seems like a few of your trace widths are kind of arbitrarily large, such as MCLR. nit a biggie, but might be good to review all of your widths and make sure they all make sense.
5. you have silkscreen over pads here: 
image
I haven't noticed it anywhere else but keep an eye out.
6. nitpicky, but noting the polarity of the battery connections is always nice.
7. I'm very confused by this connector, It doesn't match the live telemetry CAN connector, so what attaches to it?
8. Given this isn't hyper sensitive analog circuitry I'd prefer you don't daisy chain through decoupling caps as it reduces reparability / robustness, you do it in a few places 
image

9. I would add dimensions to the mounting holes, and triple check the spacing for everything is correct on the mech side of things

**Review by QuantumManiac — CHANGES_REQUESTED** [review]
image
It seems that there isn't any extension line for the dimension in the middle that measures to the hole at the bottom. Additionally, at the left and right, it would be nice if the dimensions were collinear. Example of what I mean below:  
image

As for the gerbers, they seem to be good to go if there aren't any tweaks left to be made on the PCB.

**Review by LanaTomlin — CHANGES_REQUESTED** [review]
Little bit of bookeeping, can we rename this to "papa_board" before we try and get this merged to master?

**Review by QuantumManiac — APPROVED** [review]
This board is ordered too, so we should be fine to merge to master.
